cmd/deploy/deploy.go

```diff
@@ -408,3 +416,199 @@ func (d *deployCommand) getResourceStreamClient(
+			} else {
+				jobModificationSuccess++
+				if d.verbose {
+					d.logger.Info(fmt.Sprintf("[%s] info '%s': job modified", namespaceName, resp.GetJobName()))
+				}
+			}
+
+		case models.ProgressTypeJobDeploymentRequestCreated:
+			// give summary of resolve dependency
+			if len(resolveDependencyErrors) > 0 {
+				d.logger.Error(fmt.Sprintf("[%s] Resolved dependencies of %d/%d modified jobs.", namespaceName, resolveDependencySuccess, resolveDependencySuccess+resolveDependencyFailed))
+				for _, reqErr := range resolveDependencyErrors {
+					d.logger.Error(reqErr)
+				}
+			} else {
+				d.logger.Info(fmt.Sprintf("[%s] Resolved dependency of %d modified jobs.", namespaceName, resolveDependencySuccess))
+			}
+
+			// give summary of job deletion
+			totalJobDeletionAttempt := jobDeletionSuccess + jobDeletionFailed
+			if totalJobDeletionAttempt > 0 {
+				if len(jobDeletionErrors) > 0 {
+					d.logger.Error(logger.ColoredError("[%s] Deleted %d/%d jobs.", namespaceName, jobDeletionSuccess, totalJobDeletionAttempt))
+					for _, reqErr := range jobDeletionErrors {
+						d.logger.Error(reqErr)
+					}
+				} else {
+					d.logger.Info(fmt.Sprintf("[%s] Deleted %d jobs", namespaceName, jobDeletionSuccess))
+				}
+			}
+
+			// give summary of job creation
+			totalJobCreationAttempt := jobCreationSuccess + jobCreationFailed
+			if totalJobCreationAttempt > 0 {
+				if len(jobCreationErrors) > 0 {
+					d.logger.Error(logger.ColoredError("[%s] Created %d/%d jobs.", namespaceName, jobCreationSuccess, totalJobCreationAttempt))
+					for _, reqErr := range jobCreationErrors {
+						d.logger.Error(reqErr)
+					}
+				} else {
+					d.logger.Info(fmt.Sprintf("[%s] Created %d jobs", namespaceName, jobCreationSuccess))
+				}
+			}
+
+			// give summary of job modification
+			totalJobModificationAttempt := jobModificationSuccess + jobModificationFailed
+			if totalJobModificationAttempt > 0 {
+				if len(jobModificationErrors) > 0 {
+					d.logger.Error(logger.ColoredError("[%s] Modified %d/%d jobs.", namespaceName, jobModificationSuccess, totalJobModificationAttempt))
+					for _, reqErr := range jobModificationErrors {
+						d.logger.Error(reqErr)
+					}
+				} else {
+					d.logger.Info(fmt.Sprintf("[%s] Modified %d jobs", namespaceName, jobModificationSuccess))
+				}
+			}
+
+			if !resp.GetSuccess() {
+				d.logger.Error(logger.ColoredError("[%s] Unable to request job deployment: %s", namespaceName, resp.GetValue()))
+				return "", nil
+			}
+
+			d.logger.Info(fmt.Sprintf("[%s] Deployment request created with ID: %s", namespaceName, resp.GetValue()))
+
+			return resp.Value, nil
+
+		default:
+			if d.verbose {
+				// ordinary progress event
+				d.logger.Info(fmt.Sprintf("[%s] info '%s': %s", namespaceName, resp.GetJobName(), resp.GetValue()))
+			}
+		}
+	}
+	return "", nil
+}
+
+func (d *deployCommand) pollJobDeployment(ctx context.Context, jobSpecService pb.JobSpecificationServiceClient, deployID string) error {
```

Comment:
this polling too will be common for all job deployments in all commands like job refresh lets reuse